File: csharp/bob/Bob.cs
using System.Text.RegularExpressions;

public class Bob
{
	public static string Hey(string s)
	{
		if (string.IsNullOrEmpty(s.Trim()))
		{
			return "Fine. Be that way!";
		}

		if (Regex.IsMatch(s, "[a-zA-Z]") && s.Equals(s.ToUpper()))
		{
			return "Whoa, chill out!";
		}

		if (s.Trim().EndsWith("?", System.StringComparison.CurrentCulture))
		{
			return "Sure.";
		}

		return "Whatever.";
	}
}

File: elixir/bob/bob.exs
defmodule Bob do
  defp is_all_caps(str) do
    String.upcase(str) == str and String.match?(str, ~r/\p{L}/)
  end

  defp is_question(str) do
   String.ends_with?(str, "?") 
  end

  defp is_pause(str) do
    String.trim(str) == ""
  end

  def hey(input) do
    cond do
        is_question(input) -> "Sure."
        is_pause(input) -> "Fine. Be that way!"
        is_all_caps(input) -> "Whoa, chill out!"
        true -> "Whatever."
    end
  end
end
